src/storage.h

```diff
@@ -0,0 +1,226 @@
+  inline void in_recipe_(std::string c) { in_recipe = c; }
+  inline std::string in_recipe_() const { return in_recipe; }
+
+  /// @brief current maximum amount that can be added to processing
+  inline double current_capacity() const { 
+    return (max_inv_size - processing.quantity() - stocks.quantity()); }
+
+  /// @brief the batch handling identifier
+  inline void batch_handling_(bool c) { batch_handling = c; }
+  inline bool batch_handling_() const { return batch_handling; }
+
+  /// @brief returns the time key for ready materials
+  int ready_time(){ return context()->time() - residence_time; }
+
+ protected:
+  ///   @brief adds a material into the incoming commodity inventory
+  ///   @param mat the material to add to the incoming inventory.
+  ///   @throws if there is trouble with pushing to the inventory buffer.
+  void AddMat_(cyclus::Material::Ptr mat);
+
+  /// @brief Move all unprocessed inventory to processing
+  void BeginProcessing_();
+
+  /// @brief Move as many ready resources as allowable into stocks
+  /// @param cap current throughput capacity 
+  void ProcessMat_(double cap);
+
+  /// @brief any ready resources remaining in processing get pushed off to next timestep
+  /// @param time the timestep whose buffer remains unprocessed 
+  // void AdvanceUnconverted_(int time);
+
+  /// @brief move ready resources from processing to ready at a certain time
+  /// @param time the time of interest
+  void ReadyMatl_(int time);
+
+  /* --- Module Members --- */
+
+  #pragma cyclus var {"tooltip":"input commodity",\
+                      "doc":"commodity accepted by this facility"}
+  std::vector<std::string> in_commods;
+
+  #pragma cyclus var {"default": [1]}
+  std::vector<double> in_commod_prefs;
+
+  #pragma cyclus var {"tooltip":"output commodity",\
+                      "doc":"commodity produced by this facility"}
+  std::vector<std::string> out_commods;
+
+  #pragma cyclus var {"default":"",\
+                      "tooltip":"input recipe",\
+                      "doc":"recipe accepted by this facility"}
+  std::string in_recipe;
+
+  #pragma cyclus var {"default": 0,\
+                      "tooltip":"residence time (timesteps)",\
+                      "doc":"the minimum holding time for a received commodity (timesteps)."}
+  int residence_time;
+
+  #pragma cyclus var {"default": 1e299,\
+                     "tooltip":"throughput per timestep (kg)",\
+                     "doc":"the max amount that can be processed per timestep (kg)"}
+  double throughput;
+
+  #pragma cyclus var {"default": 1e299,\
+                      "tooltip":"maximum inventory size (kg)",\
+                      "doc":"the amount of material that can be in storage"}
+  double max_inv_size; 
+
+  #pragma cyclus var {"default": True,\
+                      "tooltip":"Bool to determine how Storage handles batches",\
+                      "doc":"Determines if Storage will divide resource objects {discrete} or {continuous}. "\
```

Comment:
"If true, batches are handled as discrete quanta, neither split nor combined.  Otherwise, batches may be divided during processing."